lib/internal/modules/esm/hooks.js

```diff
@@ -143,42 +155,74 @@ class Hooks {
    * @param {string} urlOrSpecifier
    * @param {string} parentURL
    * @param {any} [data] Arbitrary data to be passed from the custom
+   * @param {number} [chainId]
+   * @param {number} [threadId]
+   * @param {any} [workerData]
    * loader (user-land) to the worker.
    */
-  async register(urlOrSpecifier, parentURL, data) {
+  async register(urlOrSpecifier, parentURL, data, chainId, threadId, workerData) {
     const cascadedLoader = require('internal/modules/esm/loader').getOrInitializeCascadedLoader();
     const keyedExports = await cascadedLoader.import(
       urlOrSpecifier,
       parentURL,
       kEmptyObject,
     );
-    await this.addCustomLoader(urlOrSpecifier, keyedExports, data);
+    await this.addCustomLoader(urlOrSpecifier, keyedExports, data, chainId, threadId, workerData);
   }
 
+
   /**
    * Collect custom/user-defined module loader hook(s).
    * @param {string} url Custom loader specifier
    * @param {Record<string, unknown>} exports
-   * @param {any} [data] Arbitrary data to be passed from the custom loader (user-land)
-   * to the worker.
+   * @param {any} [data] Arbitrary data to be passed from the custom loader (user-land) to the worker.
+   * @param {number} [chainId]
+   * @param {number} [threadId]
+   * @param {any} [workerData]
    * @returns {any | Promise<any>} User data, ignored unless it's a promise, in which case it will be awaited.
    */
-  addCustomLoader(url, exports, data) {
+  addCustomLoader(url, exports, data, chainId, threadId, workerData) {
     const {
       initialize,
       resolve,
       load,
     } = pluckHooks(exports);
 
+    if (chainId) {
```

Comment:
Can the chain ID be 0? If so this branch wouldn't execute for it.

Reply:
It was when I initially wrote the code, now it can't anymore. Removed the if condition.